#! /bin/bash
# Declare an array of strings to have a list of genomes to get
declare -a getArray=( "SAMN04957655" "SAMN04450793" "SAMN04450798" "SAMN04450799" "SAMN04450784" "SAMN04450801" "SAMN04450779" "SAMN04450778" "SAMN04450794" "SAMN03775200" "SAMN03850565" "SAMN03775199" "SAMN03020219" "SAMN03074613" "SAMN03074612" "SAMN03074621" "SAMN03074619" "SAMN01923201" "SAMN01923199" "SAMN01923198" "SAMN01888452" "SAMN01888438" "SAMN01923196" "SAMN01923192" "SAMN01923191" "SAMN01923189" "SAMN01923183" "SAMN01923177" "SAMN01923174" "SAMN01919312" "SAMN01923172" "SAMN01923169" "SAMN01923167" "SAMN01923166" "SAMN01923164" "SAMN01923162" "SAMN01923154" "SAMN01923153" "SAMN01923137" "SAMN01923136" "SAMN01923130" "SAMN01923121" "SAMN01923120" "SAMN01923118" "SAMN01923115" "SAMN01923113" "SAMN01888288" "SAMN01888442" "SAMN01923202" "SAMN01823494" "SAMN04961133" "SAMN04450796" "SAMN04450780" "SAMN04450781" "SAMN04450782" "SAMN04450783" "SAMN04450786" "SAMN04450787" "SAMN04450790" "SAMN04450795" "SAMN04450800" "SAMN04450802" "SAMN04450804" "SAMN04450808" "SAMN04450809" "SAMN04450810" "SAMN03980008" "SAMN03850561" "SAMN03852603" "SAMN03850562" "SAMN03074611" "SAMN04450789" "SAMN04450797" "SAMN04450792" "SAMN04450788" "SAMN04450785" "SAMN04450777" "SAMN04450791" "SAMN04450813" "SAMN04450807" "SAMN04450812" "SAMN04450811" "SAMN03850564" "SAMN01923119" "SAMN01923117" "SAMN01923116" "SAMN01923126" "SAMN01923127" "SAMN01923128" "SAMN01923131" "SAMN01923132" "SAMN01923134" "SAMN01923135" "SAMN01923138" "SAMN03850563" "SAMN01923159" "SAMN01923158" "SAMN01923140" "SAMN01923155" "SAMN01923143" "SAMN01923144" "SAMN01923149" "SAMN01923152" "SAMN01923156" "SAMN01923157" "SAMN01923160" "SAMN01923163" "SAMN01923165" "SAMN01923168" "SAMN01923171" "SAMN01923178" "SAMN01923179" "SAMN01923180" "SAMN01923181" "SAMN01923184" "SAMN01923187" "SAMN01923195" "SAMN01923197" "SAMN01923200" "SAMN01888441" "SAMN01888448" "SAMN01888451" "SAMN01888287" "SAMN01888289" "SAMN01888290" "SAMN01888437" "SAMN03074627" "SAMN03153479" "SAMN01923194" "SAMN01923193" "SAMN01923190" "SAMN01923188" "SAMN01923186" "SAMN01923185" "SAMN01923182" "SAMN01923176" "SAMN01923175" "SAMN01923173" "SAMN01923170" "SAMN01923161" "SAMN01923151" "SAMN01923150" "SAMN01923148" "SAMN01923147" "SAMN01923146" "SAMN01923145" "SAMN01923142" "SAMN01923133" "SAMN01923129" "SAMN01923125" "SAMN01923123" "SAMN01923114" "SAMN01888449" "SAMN01915326" "SAMN01888434" "SAMN01888435" "SAMN01888436" "SAMN01888439" "SAMN01888440" "SAMN01888443" "SAMN01888445" "SAMN01888446" "SAMN01888450" "SAMEA2757763" "SAMEA2162049" )

# Useful for printing progress
full=${#getArray[@]}
i=1

# Iterate the get array using for loop
for val in ${getArray[@]}; do
  ./enaGroupGet -d /raid/weheartbioinformatics/data -f fastq $val
  echo Finished $i of $full
  let "i+=1"
done
